### DISCUSSION TITLE
Hard requirement for deduplication

### DISCUSSION THREAD

**Discussion by shleeable** [juicedata/juicefs#3249] [General]
**What would you like to be added**:
File dedupe.

**Why is this needed**:
I run a mastodon instance and a few other fediverse instances... I want to host my media on a S3 compatible service like juicefs... but I cannot use it ATM without dedupe.

I'm mostly hosting 1KB-15MB files videos/photos/audio... and with how the mastodon instances work, they download/mirror the same files over and over.

so If I had 4 instances, i'd have 4 copies of the same file.

**Comment by SandyXSD** [comment]
For now it's not on our roadmap. You may refer to these discussions for more details:
- https://github.com/juicedata/juicefs/discussions/330
- https://github.com/juicedata/juicefs/discussions/2129

If the 4 mastodon instances share the same JuiceFS volume, it would be faster on the application level to use hardlink to do file dedup.
